pkg/abstractions/image/build.go

```diff
@@ -321,6 +324,32 @@ func (b *Builder) genContainerId() string {
 	return fmt.Sprintf("%s%s", types.BuildContainerPrefix, uuid.New().String()[:8])
 }
 
+func (b *Builder) checkForInitialImagePullError(containerId string, outputChan chan common.OutputMsg) {
+	// If the initial image pull of custom base image fails, the container will never receive
+	// a container address for the runc client to connect to.
+	// If it fails without a container address, we can still look up the exit code
+	// to terminate the build and notify the user.
+	for {
+		_, err := b.containerRepo.GetContainerAddress(containerId)
+		if err == nil {
+			return
+		}
+
+		_, err = b.containerRepo.GetContainerExitCode(containerId)
+		if err == nil {
+			outputChan <- common.OutputMsg{Done: true, Success: false, Msg: "Base image failed to load.\n"}
```

Comment:
I think we need to check if the exit code is > 0 because it looks like terminateContainer() calls SetContainerExitCode(). We also pass a reference of the exit code to terminateContainer(), which can be set from s.runcHandle.Run().